external/vcm/vcm/data_transform.py

```diff
@@ -0,0 +1,117 @@
+import dataclasses
+from toolz.functoolz import curry
+from typing import Any, Callable, Literal, Mapping, MutableMapping, Sequence, Set
+import xarray as xr
+import vcm
+
+
+DATA_TRANSFORM_REGISTRY: MutableMapping[str, Mapping[str, Any]] = {}
+
+TransformName = Literal[
+    "Q1_from_Qm_Q2",
+    "Qm_from_Q1_Q2",
+    "Q1_from_Qm_Q2_temperature_dependent",
+    "Qm_from_Q1_Q2_temperature_dependent",
+    "Q1_from_dQ1_pQ1",
+    "Q2_from_dQ2_pQ2",
+]
```

Comment:
This isn't mypy-checkable since our inputs are in yaml files, so if it goes out of date nothing will break. Because of that I'd suggest adding a unit test to make sure it stays up-to-date. You can check the values allowed in the Literal (`TransformName.__args__`) are identical to DATA_TRANSFORM_REGISTRY.keys().

The other option is to delete this type hint so it can't go out of date, `str` would be OK. But this is pretty nice documentation.

Reply:
This is already tested by `test_all_registered_transforms_are_added_to_data_transform_name_type`. If a user registers a new transform without adding its name to `TransformName`, that test will fail. Which is good, since otherwise they would not be able to use their transform out in the wild (dacite would complain).